src/animation/mod.rs

```diff
@@ -49,6 +50,7 @@ impl Default for PlayingAnimation2D {
             repeat: Default::default(),
             speed: 1.0,
             elapsed: 0.0,
+            duration: 1.0,
```

Comment:
I assume this was intentionally set to 1.0, so you are not running into issues on your side dividing by 0?
Don't really like either :D

Reply:
Yep, I changed this now to be 0.0, if we don't like it that the internal state represents no duration set as 0.0 then we can also wrap this in an Option.

Reply:
I think the field itself should be an Option.

Reply:
yeah fair, forced pushed the changes.